straxen/plugins/peaks/peak_subtyping.py

```diff
@@ -0,0 +1,621 @@
+            base_times[S1_indices], s1_s2_window, limitation)
+        pS2_containers = limited_containers(
+            base_times[pS2_indices], after_s2_window_ext, limitation)
+
+        # extra check, will be deleted after debugging
+        for i_a, a in enumerate([loneS1_containers, sloneS1_containers, S1_containers, pS2_containers]):
+            for i_b, b in enumerate([loneS1_containers, sloneS1_containers, S1_containers, pS2_containers]):
+                if i_a == i_b or len(a) == 0 or len(b) == 0:
+                    continue
+                r = strax.touching_windows(a, b)
+                assert (r[:, 1] - r[:, 0]).max() == 0, ''\
+                    + 'triggering containers should not be overlapping'
+
+        _peaks = peaks.copy()
+        _peaks['time'] = _peaks['center_time']
+        # touchng_windows needs non-zero length of things
+        _peaks['endtime'] = _peaks['time'] + 1
+
+        # assign loneS1's & sloneS1's following peaks
+        tw_after_s1 = strax.touching_windows(_peaks, loneS1_containers)
+        S1PH_indices = combine_indices(tw_after_s1)
+        mask[S1PH_indices] = PeakSubtyping.S1PH
+        tw_after_s1 = strax.touching_windows(_peaks, sloneS1_containers)
+        slS1PH_indices = combine_indices(tw_after_s1)
+        mask[slS1PH_indices] = PeakSubtyping.slS1PH
+
+        # assign S1's following peaks
+        tw_after_s1_before_ps2 = strax.touching_windows(_peaks, S1_containers)
+        _S1_small_s2s_thresholds = np.vstack(
+            [
+                np.full(len(S1_containers), large_s2_threshold),
+                # here the areas are fractions of potential pS2s, but not the real pS2s
+                # because some pS2s are filtered out
+                other_large_s2_fac * ps2_area[~lone_s1],
+            ]
+        ).max(axis=0)
+        S1PH_indices, S1olS2_indices = combine_indices_ref(
+            tw_after_s1_before_ps2, base_areas, _S1_small_s2s_thresholds)
+        mask[S1PH_indices] = PeakSubtyping.S1PH
+        mask[S1olS2_indices] = PeakSubtyping.S1olS2
+
+        # assign pS2's following peaks
+        tw_after_s1_after_ps2 = strax.touching_windows(_peaks, pS2_containers)
+        _pS2_small_s2s_thresholds = np.vstack(
+            [
+                np.full(len(pS2_containers), large_s2_threshold),
+                other_large_s2_fac * _s2_area[_ps2_max_indices],
+            ]
+        ).max(axis=0)
+        S2PH_indices, S2olS2_indices = combine_indices_ref(
+            tw_after_s1_after_ps2, base_areas, _pS2_small_s2s_thresholds)
+        mask[S2PH_indices] = PeakSubtyping.S2PH
+        mask[S2olS2_indices] = PeakSubtyping.S2olS2
+
+        # assign loneS1, S1, sloneS2, pS2
+        mask[loneS1_indices] = PeakSubtyping.loneS1
+        mask[sloneS1_indices] = PeakSubtyping.sloneS1
+        mask[S1_indices] = PeakSubtyping.S1
+        mask[pS2_indices] = PeakSubtyping.pS2
+        return mask
+
+    @staticmethod
+    def mark_other_s2s(
```

Comment:
I have not checked in very much detail, but to me it looks like that 70 % of this code is the same as above. If this is the case can we reduce the copy/past a bit.